apps/typeform/frontend/src/Field/TypeFormField.tsx

```diff
@@ -0,0 +1,254 @@
+        let selectedForm = initialState.selectedForm;
+        if (value) {
+          sdk.field.setValue(value);
+          selectedForm = (forms as FormOption[]).find(form => form.href === value)!;
+        } else {
+          selectedForm = initialState.selectedForm;
+          sdk.field.removeValue();
+        }
+        return { ...state, value, selectedForm, hasStaleData: false };
+      }
+      case ACTION_TYPES.RESET: {
+        sdk.field.removeValue();
+        return {
+          ...state,
+          value: '',
+          hasStaleData: false,
+          selectedForm: initialState.selectedForm
+        };
+      }
+      case ACTION_TYPES.UPDATE_TOKEN: {
+        const { token } = action.payload;
+        return {
+          ...state,
+          loading: true,
+          error: false,
+          token
+        };
+      }
+      case ACTION_TYPES.ERROR: {
+        return { ...state, loading: false, error: true };
+      }
+      default:
+        return state;
+    }
+  }
+
+  useEffect(() => {
+    const fetchForms = async () => {
+      try {
+        const response = (await (
+          await fetch(`${process.env.LAMBDA_ENDPOINT}/forms/${workspaceId}/${token}`)
+        ).json()) as TypeFormResponse;
+        const normalizedForms = normalizeFormResponse(response);
+        dispatch({
+          type: ACTION_TYPES.INIT,
+          payload: {
+            forms: normalizedForms
+          }
+        });
+      } catch (error) {
+        console.error(error);
+        dispatch({ type: ACTION_TYPES.ERROR });
+      }
+    };
+    fetchForms();
+    // Start auto resizer to adjust field height
+    sdk.window.startAutoResizer();
+  }, [token]);
+
+  const onChange = (event: any) => {
+    const value = event.currentTarget.value;
+    dispatch({ type: ACTION_TYPES.UPDATE_VALUE, payload: { value, forms } });
+  };
+
+  const openDialog = async () => {
+    await sdk.dialogs.openCurrentApp({
+      width: 1000,
+      parameters: {
+        value
+      },
+      title: 'Form Preview',
+      shouldCloseOnEscapePress: true,
+      shouldCloseOnOverlayClick: true
+    });
+  };
+
+  const normalizeFormResponse = (typeFormResponse: TypeFormResponse): FormOption[] => {
+    return typeFormResponse.forms.items.map(form => ({
+      name: form.title,
+      href: form._links.display,
+      id: form.id,
+      isPublic: form.settings.is_public
+    }));
+  };
+
+  if (!isUserAuthenticated()) {
+    return (
+      <TypeformOAuth
+        data-test-id="typeform-auth"
+        sdk={sdk as AppExtensionSDK}
+        expireSoon={false}
+        setToken={(token: string) =>
+          dispatch({ type: ACTION_TYPES.UPDATE_TOKEN, payload: { token } })
+        }
+      />
+    );
+  }
+
+  if (loading) {
+    return null;
+  }
+
+  if (error) {
+    return (
+      <Note noteType="negative">
+        We could not fetch your typeforms. Please make sure you are using a valid workspace ID and
+        access token.
```

Comment:
I would argue that asking the user to check if they are using a valid access token is too confusing. Most users won't understand what that even means.